Purchases/Identity/CustomerInfoManager.swift

```diff
@@ -174,9 +166,35 @@ class CustomerInfoManager {
         }
     }
 
+    @available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.0, *)
+    var customerInfoStream: AsyncStream<CustomerInfo> {
+        return AsyncStream(bufferingPolicy: .bufferingNewest(1)) { continuation in
+            let disposable = self.monitorChanges { continuation.yield($0) }
+
+            continuation.onTermination = { @Sendable _ in disposable() }
+        }
+    }
+
+    private var customerInfoChanges: [Int: (CustomerInfo) -> Void] = [:]
+
+    /// Allows monitoring changes to the active `CustomerInfo`.
+    /// - Returns: closure that removes the created observation.
+    /// - Note: this method is not thread-safe.
+    func monitorChanges(_ changes: @escaping (CustomerInfo) -> Void) -> () -> Void {
+        let index = self.customerInfoChanges.keys
```

Comment:
i'm definitely not fully understanding everything here, so feel free to ignore me... but it looks like we have a a map that we always need sorted and the keys don't really hold a value we care about...could we just use a queue or stack to save on space and sorting time?

Reply:
The issue is that we need to return a closure that removes the observation, without affecting the other ones. Array indexes aren't maintained when you remove an item.
Let's say we have 3 observations: 0, 1, 2.
If #1 wants to stop it (because the `AsyncStream` is cancelled for example), if we were to remove it from the array, number 2 would become `1`, which would break.

Not sure if that makes sense, or maybe you have another idea how to do this better. I don't like this solution much either :/ but it was the only option like I explained in the PR AFAIK.

Reply:
ah ok, that's tricky. in the documentation the QuakeMonitor example might be something to consider? 

so basically having a `CustomerInfoMonitor` class/protocol with the (CustomerInfo -> Void) callback, plus a method to start and stop monitoring? starting monitoring would be adding itself to a list of "monitors" in the `CustomerInfoManager`, stopping would be removing itself. 

then this Stream would be reduced to just calling `yield(monitor.callback)` and `continuation.onTermination {@Sendable _ in monitor.stopMonitoring()}` on the list of monitors

my head is a little scrambled looking at this though, i can't tell if that would work..

Reply:
I actually started with that `QuakeMonitor` idea from the docs hehe, and this is what I arrived to.

Reply:
maybe we should document the reasoning behind using a dict here

Reply:
Oh sorry I missed that: #1145